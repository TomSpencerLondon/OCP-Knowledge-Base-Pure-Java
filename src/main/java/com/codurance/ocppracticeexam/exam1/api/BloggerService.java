package com.codurance.ocppracticeexam.exam1.api;

public class BloggerService {
}
