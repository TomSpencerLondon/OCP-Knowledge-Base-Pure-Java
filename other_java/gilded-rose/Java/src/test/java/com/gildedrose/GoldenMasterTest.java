package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import java.util.stream.IntStream;

public class GoldenMasterTest {

    @Test
    void gilded_rose_should() {
        Item[] items = new Item[]{
            new Item("+5 Dexterity Vest", 10, 20), //
            new Item("Normal Item", -1, 2), //
            new Item("Aged Brie", 2, 0), //
            new Item("Aged Brie", -1, 0), //
            new Item("Elixir of the Mongoose", 5, 7), //
            new Item("Sulfuras, Hand of Ragnaros", 0, 80), //
            new Item("Sulfuras, Hand of Ragnaros", -1, 80),
            new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 48),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 47),
            new Item("Backstage passes to a TAFKAL80ETC concert", -1, 47),
            // this conjured item does not work properly yet
            new Item("Conjured Mana Cake", 3, 6)};

        GildedRose app = new GildedRose(items);
        IntStream.range(0, 50)
            .forEach((int i) -> app.updateQuality());

        Approvals.verifyAll("", items);
    }
}
