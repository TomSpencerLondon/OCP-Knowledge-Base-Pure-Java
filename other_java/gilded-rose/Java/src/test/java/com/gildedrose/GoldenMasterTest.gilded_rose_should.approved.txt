[00] = +5 Dexterity Vest, -40, 0
[01] = Normal Item, -51, 0
[02] = Aged Brie, -48, 50
[03] = Aged Brie, -51, 50
[04] = Elixir of the Mongoose, -45, 0
[05] = Sulfuras, Hand of Ragnaros, 0, 80
[06] = Sulfuras, Hand of Ragnaros, -1, 80
[07] = Backstage passes to a TAFKAL80ETC concert, -35, 0
[08] = Backstage passes to a TAFKAL80ETC concert, -40, 0
[09] = Backstage passes to a TAFKAL80ETC concert, -45, 0
[10] = Backstage passes to a TAFKAL80ETC concert, -45, 0
[11] = Backstage passes to a TAFKAL80ETC concert, -45, 0
[12] = Backstage passes to a TAFKAL80ETC concert, -51, 0
[13] = Conjured Mana Cake, -47, 0
